### PR TITLE
Add member offsets and array strides

### PR DESCRIPTION
**Pull request by kvark**
Related to:
- https://github.com/gpuweb/gpuweb/issues/561
- https://github.com/gpuweb/gpuweb/issues/773

### CODE REVIEW

File: src/front/spv.rs [hunk 1]
```diff
@@ -1208,14 +1237,23 @@ impl<I: Iterator<Item = u32>> Parser<I> {
...
+                    .ok_or(Error::MissingDecoration(spirv::Decoration::Offset))?,
             });
         }
         let inner = crate::TypeInner::Struct { members };
+        let decor = self.future_decor.remove(&id);
+        if let Some(ref decor) = decor {
+            if decor.block.is_some() {
+                // do nothing
+            }
+        }
```
Comment (Napokue) [inline src/front/spv.rs:1251]: What is the reason for doing this, as the code does not do anything with it?

File: src/back/msl.rs [hunk 2]
```diff
@@ -451,7 +451,11 @@ impl<W: Write> Writer<W> {
                         write!(self.out, ".{}", COMPONENTS[index as usize])?;
                         Ok(MaybeOwned::Owned(crate::TypeInner::Scalar { kind, width }))
                     }
-                    crate::TypeInner::Array { base, size } => {
+                    crate::TypeInner::Array {
+                        base,
+                        size,
+                        stride: _,
```
Comment (Napokue) [inline src/back/msl.rs:457]: Does Metal have a stride or is it just that it is not planned yet to support strides for our Metal back-end?

File: src/front/spv.rs (hunk 1 above)
Comment (kvark) [reply src/front/spv.rs:1251]: the decoration applies to that type, so it needs to be used here, somehow
I just haven't figured out how exactly :)

File: src/back/msl.rs (hunk 2 above)
Comment (kvark) [reply src/back/msl.rs:457]: we'll need to insert padding ourselves, that's a big TODO for now

**Pull request merged**